cpp/src/io/orc/reader_impl.cu

```diff
@@ -213,16 +204,15 @@ size_t gather_stream_info(const size_t stripe_index,
     }
     if (col != -1) {
       if (src_offset >= stripeinfo->indexLength || use_index) {
-        // NOTE: skip_count field is temporarily used to track index ordering
-        auto& chunk = chunks[stripe_index][col];
-        const auto idx =
-          get_index_type_and_pos(stream.kind, chunk.skip_count, col == orc2gdf[column_id]);
-        if (idx.first < gpu::CI_NUM_STREAMS) {
-          chunk.strm_id[idx.first]  = stream_info.size();
-          chunk.strm_len[idx.first] = stream.length;
-          chunk.skip_count          = idx.second;
-
-          if (idx.first == gpu::CI_DICTIONARY) {
+        auto& chunk           = chunks[stripe_index][col];
+        auto const index_type = get_stream_index_type(stream.kind);
+        if (index_type < gpu::CI_NUM_STREAMS) {
+          chunk.strm_id[index_type]  = stream_info.size();
+          chunk.strm_len[index_type] = stream.length;
+          // NOTE: skip_count field is temporarily used to track the presence of index streams
+          chunk.skip_count |= 1 << index_type;
```

Comment:
Is the `skip_count` now equivalent to what was being computed before? This change seems separate from the stream ordering, but also the skip_count looks like it would be subtly different now from before.

Reply:
Yes, `skip_count` value changed. It used to hold order of streams within a column. Each stream type was held in a byte of the skip_count value. Since we don't need the order during decode, I replaced this with a bitmap that uses stream type as bit "index".
In either case, this is a temporary (ab)use of this data member. I'm not a fan of this approach but did not change it in this PR, as I want to keep the changes minimal (this is a risky PR IMO). We can replace this reuse of `skip_count` in a future PR.